frontend/subscriptions/index.js

```diff
@@ -0,0 +1,45 @@
+import {
+  fetchProductRelations,
+  cachedProductReceived$,
+  productReceived$,
+} from '@shopgate/engage/product';
+import getConfig from '../helpers/getConfig';
+
+const { productPage, productPageAddToCart } = getConfig();
+
+/**
+ * @param {Function} subscribe subscribe
+ */
+export default (subscribe) => {
+  const processProduct$ = productReceived$.merge(cachedProductReceived$);
+  subscribe(processProduct$, ({ action, dispatch }) => {
+    const { id: productId, baseProductId } = action.productData;
+
+    if (productPage.type) {
+      dispatch(fetchProductRelations({
+        productId,
+        type: productPage.type,
+      }));
+
+      if (baseProductId) {
```

Comment:
what if productId and baseProductId are the same? will it not cause two request?

Reply:
baseProductId is null in this case and if is evaluated to false